plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -515,15 +538,10 @@ runCekM (MachineParameters costs runtime) (ExBudgetMode getExBudgetInfo) (Emitte
     logs <- _cekEmitterInfoGetFinal
     pure (errOrRes, st, logs)
 
--- | Extend an environment with a variable name, the value the variable stands for
--- and the environment the value is defined in.
-extendEnv :: Name -> CekValue uni fun -> CekValEnv uni fun -> CekValEnv uni fun
-extendEnv = insertByName
-
 -- | Look up a variable name in the environment.
-lookupVarName :: forall uni fun s . (PrettyUni uni fun) => Name -> CekValEnv uni fun -> CekM uni fun s (CekValue uni fun)
-lookupVarName varName varEnv =
-    case lookupName varName varEnv of
+lookupVarName :: forall uni fun s . (PrettyUni uni fun) => NamedDeBruijn -> CekValEnv uni fun -> CekM uni fun s (CekValue uni fun)
+lookupVarName varName@(NamedDeBruijn _ varIx) varEnv =
+    case varEnv `Env.index` fromIntegral varIx of
```

Comment:
My *guess* is that constantly hitting this `fromIntegral` is going to cost us, since it will do actual work.

Reply:
Suggestion: make `Index` a `Word` and don't have `Level` be a wrapper around `Index` (possibly then need to take some care in the debruijn passes).